src/super_gradients/training/models/pose_estimation_models/yolo_nas_pose.py

```diff
@@ -0,0 +1,239 @@
+
+        self.offset_transition = ConvBNAct(
+            inter_channels, channels_per_kpt * num_classes, kernel_size=3, stride=1, padding=1, bias=False, activation_type=activation_type
+        )
+
+        offset_modules = []
+        for _ in range(num_classes):
+            offset_modules.append(self.build_offset_path(inter_channels, activation_type, num_blocks, channels_per_kpt, upsample_factor))
+        self.offset_modules = nn.ModuleList(offset_modules)
+
+    def forward(self, feats: Tuple[Tensor]):
+        biggest_size = feats[0].size()[2:]
+        feats = [feats[0]] + [torch.nn.functional.interpolate(x, size=biggest_size, mode="bilinear", align_corners=False) for x in feats[1:]]
+        feats = torch.cat(feats, dim=1)
+        feats = self.stem(feats)
+
+        heatmap = self.heatmap(feats)
+        offset_features = self.offset_transition(feats)
+        final_offset = []
+
+        for j in range(self.num_classes):
+            offset_specific_features = offset_features[:, j * self.channels_per_kpt : (j + 1) * self.channels_per_kpt]
+            offset_predictions = self.offset_modules[j](offset_specific_features)
+            final_offset.append(offset_predictions)
+
+        offsets = torch.cat(final_offset, dim=1)
+        return heatmap, offsets
+
+    def build_heatmap_path(self, inter_channels, activation_type, num_blocks, num_joints, upsample_factor):
+        blocks = [
+            (
+                f"conv_bn_act_{block_index:02d}",
+                ConvBNAct(inter_channels, inter_channels, kernel_size=3, stride=1, padding=1, bias=False, activation_type=activation_type),
+            )
+            for block_index in range(num_blocks)
+        ]
+        blocks += [("final", nn.Conv2d(inter_channels, num_joints * (2**upsample_factor), kernel_size=1)), ("shuffle", nn.PixelShuffle(upsample_factor))]
+        return nn.Sequential(collections.OrderedDict(blocks))
+
+    def build_offset_path(self, inter_channels, activation_type, num_blocks, channels_per_kpt, upsample_factor):
+        blocks = []
+        blocks.extend(
+            [
+                (
+                    f"conv_bn_act_{block_index:02d}",
+                    ConvBNAct(channels_per_kpt, channels_per_kpt, kernel_size=3, stride=1, padding=1, bias=False, activation_type=activation_type),
+                )
+                for block_index in range(num_blocks)
+            ]
+        )
+
+        blocks += [("final", nn.Conv2d(channels_per_kpt, 2 * (2**upsample_factor), kernel_size=1)), ("shuffle", nn.PixelShuffle(upsample_factor))]
+        return nn.Sequential(collections.OrderedDict(blocks))
+
+    @property
+    def out_channels(self):
+        return (self.num_classes_with_center), (2 * self.num_classes)
+
+
+class YoloNASPose(CustomizableDetector):
+    def __init__(self, arch_params: Union[HpmStruct, DictConfig]):
```

Comment:
lets drop the arch_params here please.
we said we will not create new models with that concept.
This init should accept the actual parameters by name (only the variants below should accept arch_params)